src/test/java/com/alkemy/ong/TestimonialControllerFunctionalTest.java

```diff
@@ -0,0 +1,246 @@
+        Long id = withCreatedTestimonial();
+        String endpointUrl = testimonialControllerUrl + "/{id}";
+        HttpHeaders headers = new HeaderBuilder()
+                .withValidToken("admin1@gmail.com", 3600L)
+                .build();
+        entity = new HttpEntity(null, headers);
+        // When
+        ResponseEntity<ErrorDTO> response = testRestTemplate.exchange(
+                endpointUrl,
+                HttpMethod.DELETE,
+                entity,
+                new ParameterizedTypeReference<>() {
+                },
+                Map.of("id", 2)
+        );
+        assertEquals(404, response.getStatusCode().value());
+    }
+
+    @Test
+    void testUpdateTestimonial_shouldReturnOkResponse() {
+        TestimonialCreationDTO testimonialCreationDTO = createTestimonialCreationDTO();
+        Long id = withCreatedTestimonial();
+        HttpHeaders headers = new HeaderBuilder()
+                .withValidToken("admin1@gmail.com", 3600L)
+                .build();
+        TestimonialUpdateDTO testimonialUpdateDTO = TestimonialUpdateDTO.builder()
+                .name("Santiago")
+                .image("santiago.jpg")
+                .content("este es mi testimonio")
+                .build();
+        entity = new HttpEntity(testimonialUpdateDTO, headers);
+        String endpointUrl = testimonialControllerUrl + "/{id}";
+        // When
+        ResponseEntity<ErrorDTO> response = testRestTemplate.exchange(
+                endpointUrl,
+                HttpMethod.PUT,
+                entity,
+                new ParameterizedTypeReference<>() {
+                },
+                Map.of("id", id)
+        );
+        assertEquals(200, response.getStatusCode().value());
+    }
+
+    @Test
+    void testUpdateTestimonial_shouldReturnNotFoundException() {
+        TestimonialUpdateDTO testimonialUpdateDTO = TestimonialUpdateDTO.builder()
+                .name("Santiago")
+                .image("santiago.jpg")
+                .content("este es el testimonio de Santiago")
+                .build();
+        String endpointUrl = testimonialControllerUrl + "/{id}";
+
+        entity = new HttpEntity(testimonialUpdateDTO, null);
+        // When
+        ResponseEntity<ErrorDTO> response = testRestTemplate.exchange(
+                endpointUrl,
+                HttpMethod.PUT,
+                entity,
+                new ParameterizedTypeReference<>() {
+                },
+                Map.of("id", "1")
+        );
+        assertEquals(403, response.getStatusCode().value());
+    }
+
+
+    @Test
+    void testGetTestimonials_shouldReturnOkResponse() {
+        String endpointUrl = testimonialControllerUrl;
+        HttpHeaders headers = new HeaderBuilder()
+                .withValidToken("admin1@gmail.com", 3600L)
+                .build();
+        entity = new HttpEntity(null, headers);
+        ResponseEntity<TestimonialDTO> response = testRestTemplate.exchange(
```

Comment:
Este test está fallando porque `getTestimonials` te devuelve un `TestimonialListDTO` no un `TestimonialDTO`. Deberías cambiar esta clase acá y ahi ver que pasa.

Reply:
Santi lo cambie por 
ResponseEntity<?> response = testRestTemplate.exchange(
                endpointUrl,
                HttpMethod.GET,
                entity,
                new ParameterizedTypeReference<>() {
                },
                Map.of()

Reply:
Pero igual me sigue dando el 406

Reply:
No tenes que ponerle `ResponseEntity<?>` porque en este caso sí sabemos que devuelve, tenes que ponerle `ResponseEntity<TestimonialListDTO>`.
Te está tirando 406 porque a la entidad que devuelve el método, Jackson no sabe como serializarla (si te fijas en TestimonialListDTO no hay annotations). Entonces tenes que agregarle:

1. `@Data`
2. `@JsonInclude(JsonInclude.Include.NON_NULL)`
3. `@NoArgsConstructor`